crates/worker/src/executor/process.rs

```diff
@@ -40,16 +42,72 @@ impl Execution for ProcessExecution {
     }
 }
 
+struct RuntimeContext {
+    socket_path: String,
+    work_dir: String,
+    job_json: String,
+}
+
+impl RuntimeContext {
+    fn new(socket_path: String, work_dir: String, job_json: String) -> Self {
+        Self {
+            socket_path,
+            work_dir,
+            job_json,
+        }
+    }
+}
+
+fn interpolate_runtime_arg(arg: &str, ctx: &RuntimeContext) -> String {
```

Comment:
I think this would benefit from some tests, because for me this is not obvious what it is exactly doing.